packages-next/keystone/src/lib/core/field-assertions.ts

```diff
@@ -69,30 +69,12 @@ function assertIdFieldGraphQLTypesCorrect(list: ListForValidation) {
       `The idField on a list must define a uniqueWhere GraphQL input with the ID GraphQL scalar type but the idField for ${list.listKey} does not define one`
     );
   }
-  if (idField.input.uniqueWhere.arg.type !== schema.ID) {
-    throw new Error(
-      `The idField on a list must define a uniqueWhere GraphQL input with the ID GraphQL scalar type but the idField for ${
-        list.listKey
-      } defines the type ${idField.input.uniqueWhere.arg.type.graphQLType.toString()}`
-    );
-  }
-  // we may want to loosen these constraints in the future
-  if (idField.input.create !== undefined) {
-    throw new Error(
-      `The idField on a list must not define a create GraphQL input but the idField for ${list.listKey} does define one`
-    );
-  }
-  if (idField.input.update !== undefined) {
-    throw new Error(
-      `The idField on a list must not define an update GraphQL input but the idField for ${list.listKey} does define one`
-    );
-  }
```

Comment:
Those already can be controlled by `access` property of field:
```js
      access: {
        create: true,
        update: true,
      },
```
so we must not disable this.